### PR TITLE
added periodical speed change

### PR DESCRIPTION
**Pull request by vidoree**
Periodaical speed change was implemented using Timer.

### CODE REVIEW

**Review by Skyraider150 — APPROVED** [review]
The implementation is great, albeit the arbitrary 2000 value could probably use some contextual commenting. Otherwise flawless.

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java [hunk 1]
```diff
@@ -41,9 +44,17 @@ private void enforceSpeedLimit() {
...
 	public void setJoystickPosition(int joystickPosition) {
-		this.step = joystickPosition;		
+		this.step = joystickPosition;
+		Timer timer = new Timer();
+		timer.schedule(new TimerTask(){
+			@Override
+			public void run() {
+				followSpeed();
+			}
+		}, 2000);
```
Comment (Skyraider150) [inline train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java:57]: Here: this value could use some comments.

**Review by Skyraider150 — CHANGES_REQUESTED** [review]

File: train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java (hunk 1 above)
Comment (vidoree) [reply train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java:57]: It is a clear context which doesn't need any explanation.

Comment (Skyraider150) [reply train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java:57]: kk

**Pull request merged**